### PR TITLE
#166789995 Enable Social Login via Google, Facebook and Twitter

### PR DESCRIPTION
**Pull request by niyoceles**
#### What does this PR do?
- Enable login using Google, Facebook, and Twitter
#### Description of Task to be completed
- Configure passport to enable social login
- Create an application to Google, Facebook, and Twitter
#### How should this be manually tested?
1. Run the following commands in Terminal
```
git clone git@github.com:andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-enable-social-login-166789995
```
2. run `npm install`

3. Install your local database in Postgres

4. Rename `.env.sample` to `.env` Open it and replace  `database-name` your created database

5. run `npm run start`

6. Using Google Chrome (or any other browser) 
- `http://localhost:3000/api/auth/facebook`
- `http://localhost:3000/api/auth/google`
- `http://localhost:3000/api/auth/twitter`

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166789995](https://www.pivotaltracker.com/story/show/166789995)
#### Screenshots (if appropriate)
N/A

#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by elemanhillary-zz — DISMISSED** [review]
social login via social platforms work

**Review by Hervera — DISMISSED** [review]

**Review by IsaiahRn — DISMISSED** [review]
Social login enabled, it can be merged!

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]
After success the travis file should run `npm run coveralls` we need that coverage

File: tests/passport.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  //   http.get('http://localhost:3000/auth/twitter', (res) => {
+  //     assert.equal(302, res.statusCode);
+  //     done();
+  //   });
+  // });
+});
+
+describe('Login with Social media Fail', () => {
+  it('Not found Url should return 404', (done) => {
+    http.get('http://localhost:3000/auth/facebookk', (res) => {
```
Comment (Hervera) [inline tests/passport.test.js]: Change the route name to http://localhost:3000/auth/facebook

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
Fix Missing trailing comma comma-dangle and the indentation space warnings from Hound please

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
fix the comma dangle warning, and append a new line at EOF in `.travis.yml`

**Review by Hervera — DISMISSED** [review]

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
fix the `.travis.yml` file as @kodek-sleuth stated with sample screenshots on how it should be

**Review by IsaiahRn — DISMISSED** [review]

**Review by Hervera — DISMISSED** [review]

**Comment by niyoceles** [comment]
> .travis.yml

I fixed that.

**Review by kodek-sleuth — DISMISSED** [review]
This PR is ready to be merged

**Review by kodek-sleuth — DISMISSED** [review]
This PR is ready to be merged

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by elemanhillary-zz — DISMISSED** [review]
approved

**Review by kodek-sleuth — DISMISSED** [review]
This PR is ready to be merged

**Review by Hervera — DISMISSED** [review]
Ready to be marged

**Review by kodek-sleuth — APPROVED** [review]
This PR is ready to be merged

**Review by elemanhillary-zz — APPROVED** [review]
ready for merging

**Review by Hervera — APPROVED** [review]
Ready to be marged

**Review by IsaiahRn — APPROVED** [review]
ready to be merged!

**Pull request merged**